framework/src/modules/chain/logic/init_transaction.js

```diff
@@ -101,30 +117,54 @@ const outTransferAsset = raw => {
 	return { outTransfer };
 };
 
+// TODO: remove after https://github.com/LiskHQ/lisk/issues/2424
+const assetDbReadMap = new Map([
+	[0, transferAsset],
+	[1, signatureAsset],
+	[2, delegateAsset],
+	[3, voteAsset],
+	[4, multiAsset],
+	[5, dappAsset],
+	[6, inTransferAsset],
+	[7, outTransferAsset],
+]);
+
 class Transaction {
-	constructor(transactions = {}) {
+	constructor({ registeredTransactions = {} }) {
 		this.transactionClassMap = new Map();
 
-		// TODO: remove after https://github.com/LiskHQ/lisk/issues/2424
-		this.assetDbReadMap = new Map([
-			[0, transferAsset],
-			[1, signatureAsset],
-			[2, delegateAsset],
-			[3, voteAsset],
-			[4, multiAsset],
-			[5, dappAsset],
-			[6, inTransferAsset],
-			[7, outTransferAsset],
-		]);
-
-		Object.keys(transactions).forEach(transactionType => {
+		Object.keys(registeredTransactions).forEach(transactionType => {
 			this.transactionClassMap.set(
 				Number(transactionType),
-				transactions[transactionType]
+				registeredTransactions[transactionType]
 			);
 		});
 	}
 
+	fromBlock(block) {
+		if (!block.transactions) {
+			return undefined;
+		}
+
+		const transactions = block.transactions.map(transaction =>
+			this.fromJson(transaction)
+		);
+
+		return transactions;
+	}
+
+	fromJson(rawTx) {
+		const TransactionClass = this.transactionClassMap.get(rawTx.type);
+
+		if (!TransactionClass) {
+			throw new Error('Transaction type not found.');
+		}
+
+		const transaction = new TransactionClass(rawTx);
+		transaction.matcher = TransactionClass.matcher;
```

Comment:
Do we really need to assign class property to instance, would it not come along with `new TransactionClass`, it not then we should revisit the logic where we assign `matcher` to the class itself.

Reply:
No, the instance won't have access to `matcher` unless it's assigned like it's done

Reply:
above it's the difference between static function and instance method

Reply:
As I earlier said, if that's not the case we should update Transaction class in a way that whenever and wherever in the code I initiate the object I always get the matcher along. 

Where we are defining 

```js
Transaction.matcher = matcher;
```

Change it to 

```js
Object.defineProperty(TransactionClass.prototype, 'matcher', {get: ()=> matcher});
```

The idea is once you setup the class once, you don't need to remember to copy data attributes to instances manually.